internal/api/employee.go

```diff
@@ -0,0 +1,164 @@
+package api
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/SharanyaSD/Payroll-GoLang.git/internal/app/emp"
+	"github.com/SharanyaSD/Payroll-GoLang.git/internal/pkg/dto"
+)
+
+func CreateEmployeeHandler(empSvc emp.Service) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		var req dto.CreateEmployeeRequest
+		err := json.NewDecoder(r.Body).Decode(&req)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(err.Error()))
+			return
+		}
+
+		fmt.Printf("%+v", req)
+
+		if req.DateOfBirth.After(time.Now()) {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("Date of birth cannot be in the future"))
+			return
+		}
+
+		employeeInfo, err := empSvc.CreateEmployee(req)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(err.Error()))
+			return
+		}
+
+		responseJSON, err := json.Marshal(employeeInfo)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(err.Error()))
+			return
+		}
+
+		w.WriteHeader(http.StatusOK)
+		w.Write(responseJSON)
+	}
+}
+
+func DeleteEmployeeHandler(empSvc emp.Service) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		empID := r.URL.Query().Get("id")
+		if empID == "" {
```

Comment:
use a stndard format for success and failure response, something like

Failure:

```
{
    "message": "error message"
 }
```
 
 
 
 Success:
 
 
```
 {
 "data": {
 "field1": "val1",
 "field2": "val2"
 }
 }
```